core/query/src/main/java/org/eclipse/rdf4j/query/impl/BackgroundGraphResult.java

```diff
@@ -0,0 +1,244 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Eclipse RDF4J contributors, Aduna, and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.query.impl;
```

Comment:
They can't both be in the same module because the module heirarchy rio-api -> query -> queryresultio-api doesn't allow it. Ie, to do so I would need to relocate this from core/query to core/queryresultio/api where it would be the only class dealing with Statement's, as all of the others in queryresultio-api deal with BindingSet.

I can create a new helpers package in core/query/ if you prefer, but the impl class contains all other similar classes in core/query.

This relates to my comment about wanting to merge queryresultio-api into rio-api in a future major version, to remove differences like this. To do so we would need to cut the tie from queryresultio-api to query, possibly by pushing some classes into query from queryresultio-api.